src/test/java/com/google/sps/servlets/TATest.java

```diff
@@ -50,24 +54,133 @@ public void tearDown() {
   }
 
   @Test
-  // Add a TA and their class to the TA datastore
-  public void addNewTA() throws Exception {
+  // Add one TA to an empty class TA list
+  public void addOneTAEmptyList() throws Exception {
+    Entity classEntity = new Entity("Class");
+
+    classEntity.setProperty("owner", "ownerID");
+    classEntity.setProperty("name", "testClass");
+    classEntity.setProperty("beingHelped", new EmbeddedEntity());
+    classEntity.setProperty("studentQueue", Collections.emptyList());
+    classEntity.setProperty("taList", Collections.emptyList());
+    classEntity.setProperty("visitKey", "");
+
+    datastore.put(classEntity);
 
     // Create examples for the TA email and class code
     when(httpRequest.getParameter("taEmail")).thenReturn("test@google.com");
-    when(httpRequest.getParameter("classCode")).thenReturn("testClassCode");
+    when(httpRequest.getParameter("classCode"))
+        .thenReturn(KeyFactory.keyToString(classEntity.getKey()));
 
     // Firebase needs to return the correct user ID for the TA
     UserRecord userRecord = mock(UserRecord.class);
     when(authInstance.getUserByEmail("test@google.com")).thenReturn(userRecord);
-    when(userRecord.getUid()).thenReturn("taID");
 
     addTA.doPost(httpRequest, httpResponse);
 
-    Entity testEntity = datastore.prepare(new Query("TA")).asSingleEntity();
+    // Obtain the corresponding class entity and its TA list
+    Entity testEntity = datastore.get(classEntity.getKey());
+    List<String> listOfClassTAs = (List) testEntity.getProperty("taList");
+
+    // Verify the new TA was added to the class TA roster
+    assertTrue(listOfClassTAs.contains("test@google.com"));
+  }
+
+  @Test
+  // Add one TA to an empty class TA list
```

Comment:
I think this comment needs updating, same comment on multiple tests in this file